lib/index.js

```diff
@@ -94,7 +95,7 @@ module.exports = function (
     isPaused = false;
     const cmd = nodeArgs.concat(wrapper, script, scriptArgs);
 
-    if (extname(script) === '.mjs') {
+    if (extname(script) === '.mjs' || getPackageType.sync(script) === 'module') {
```

Comment:
This spot is a little more tricky because we have the child in the parent scope.

let's return the `isESM` promise promise here and I'll see how I can make it cleaner after it's merged.

Reply:
Funnily enough, this spot was the easy one. I just make `start()` `async` and `await` here, and it works, all tests pass.